### PR TITLE
Fix children type for react-native-dialog

### PR DESCRIPTION
**Pull request by ibarrae**
Fix for allowing dynamic components inside of `Dialog.Container`.

### CODE REVIEW

**Comment by ibarrae** [comment]
Hey @sestrella, could you please add Cristian Motoche to reviewers? I can't add him for a review so we can continue with this.

**Comment by sestrella** [comment]
@ibarrae - done, I added @CristhianMotoche

File: types/react-native-dialog/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Type definitions for react-native-dialog 4.0
 // Project: https://github.com/mmazzarolo/react-native-dialog
 // Definitions by: MrLuje <https://github.com/MrLuje>
+//                 Stack Builders <https://github.com/stackbuilders>
+//                 Esteban <https://github.com/ibarrae>
```
Comment (CristhianMotoche) [inline types/react-native-dialog/index.d.ts]: @ibarrae Could you please add your lastname, too?
I'm doing my review based on this [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30135/files#diff-8442333972e5d92c77013f733e8edcc7R3) and I see they write their full name there.

File: types/react-native-dialog/index.d.ts [hunk 2]
```diff
@@ -32,7 +34,7 @@ interface ButtonProps {
 
 interface ContainerProps {
     blurComponentIOS?: ReactNode;
-    children: JSX.Element[];
+    children: React.ReactNode[];
```
Comment (CristhianMotoche) [inline types/react-native-dialog/index.d.ts]: @ibarrae Is `React.ReactNode[]` the same as `React.ReactNode`? When do you use one or the other?

File: types/react-native-dialog/react-native-dialog-tests.tsx [hunk 3]
```diff
@@ -23,6 +37,7 @@ class Example extends Component {
                     label="Validate"
                     onPress={() => console.log("test")}
                 />
+                {this.dynamicButtons()}
```
Comment (CristhianMotoche) [inline types/react-native-dialog/react-native-dialog-tests.tsx]: @ibarrae Is there a tests for a single child component? 
```jsx
class Example extends Component { 
  f(){ 
     return (<Dialog.Button
                           label={label}
                           onPress={action}
     />);
}
...
{this.f()}
...
}
```

File: types/react-native-dialog/index.d.ts (hunk 2 above)
Comment (ibarrae) [reply types/react-native-dialog/index.d.ts]: You lost me here... could you rephrase the question please?

File: types/react-native-dialog/index.d.ts (hunk 1 above)
Comment (ibarrae) [reply types/react-native-dialog/index.d.ts]: Sure thing, thanks for the tip.

File: types/react-native-dialog/react-native-dialog-tests.tsx (hunk 3 above)
Comment (ibarrae) [reply types/react-native-dialog/react-native-dialog-tests.tsx]: Nope, but that's a good point, going to add a test like that also.

File: types/react-native-dialog/index.d.ts (hunk 2 above)
Comment (CristhianMotoche) [reply types/react-native-dialog/index.d.ts]: Ok, based on our conversation. It seem that this covers both cases: a single react node and a list of react nodes. Thanks for the clarification.
